### PR TITLE
adding telemetry

### LINKED ISSUE #1691: [FEA] Enable telemetry during RMM's build process
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I wish I could use RMM to do [...]
Enable telemetry during RMM's build process

**Describe the solution you'd like**
Implement OpenTelemetry, as described in https://github.com/rapidsai/build-infra/issues/139

**Additional context**
* https://github.com/rapidsai/cudf/pull/16924
* https://github.com/rapidsai/shared-workflows/pull/247
* https://github.com/rapidsai/gha-tools/pull/118

This issue is a placeholder to attach a testing PR with. No work is required on the part of the RMM maintainers, though I'm happy to discuss any thoughts or concerns you may have.

### PR DESCRIPTION
**Pull request by msarahan**
## Description
close #1691 

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml
Comment (ajschmidt8) [inline .github/workflows/pr.yaml:229]: will this job and the `telemetry-setup` eventually be moved to a shared workflow?

Comment (msarahan) [reply .github/workflows/pr.yaml:229]: probably a shared action, not a shared workflow, but yes. that might help improve the DRY stuff here.

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:17]: Use spaces around all the `${{ }}` for consistency. We use this style pretty much everywhere in our actions files.
```suggestion
  OTEL_RESOURCE_ATTRIBUTES: "git.repository=${{ github.repository }},git.ref=${{ github.ref }},git.sha=${{ github.sha }},git.job_url=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:58]: Must we clone this repository? Can we not use a "shared action" as usual?

Something like `uses: rapidsai/shared-actions/...@main`?

**Review by bdice — COMMENTED** [review]
What can we do to make this less verbose?

File: .github/workflows/pr.yaml
Comment (msarahan) [reply .github/workflows/pr.yaml:58]: Yes, we must clone it. There's an explanation for this change at https://github.com/rapidsai/shared-actions. TL;DR - this is the only way to allow shared actions to depend on other shared actions.

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:145]: pr-builder has always been the last job until now. PRs can be merged once that completes. Is it important for this to complete before merges occur?

Comment (msarahan) [reply .github/workflows/pr.yaml:145]: It was easy to piggy-back on pr-builder being the last thing so that everything else is done. It is not important for this to be complete before merges occur, but it is important that it start only after all other jobs (except pr-builder) are complete. Is there a better way to express that? Making it need conda-python-tests and wheel-python-tests would probably be good enough, and then this would run in parallel with pr-builder. I find that conceptually more messy, but I'm fine with it from a practical standpoint.

If the concern is that the merge would mess up the telemetry, I think that's also going to be OK. The telemetry is all based on github actions run metadata, and I believe that continues to exist until it ages out after some amount of time. The few seconds between a merge event and the summarize action won't matter.

Comment (bdice) [reply .github/workflows/pr.yaml:145]: Great. I am fine with this depending on pr-builder.

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:13]: We try to keep `pr-builder` at the top of the jobs list. Move this somewhere below, probably next in line since it's a dependency of many other jobs.

**Review by bdice — APPROVED** [review]
One suggestion for moving code, otherwise LGTM.

**Pull request merged**
